src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/MSBuildSdkExtrasTests.cs

```diff
@@ -0,0 +1,57 @@
+using Xamarin.ProjectTools;
+using NUnit.Framework;
+using System.IO;
+using System.Linq;
+
+namespace Xamarin.Android.Build.Tests
+{
+	[TestFixture]
+	public class MSBuildSdkExtrasTests : BaseTest
+	{
+		[Test]
+		public void ClassLibrary ()
+		{
+			var proj = new MSBuildSdkExtrasProject ();
+			using (var b = CreateDllBuilder (Path.Combine ("temp", TestName))) {
+				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
+				var output = Path.Combine (Root, b.ProjectDirectory, proj.OutputPath, proj.ProjectName + ".dll");
+				FileAssert.Exists (output);
+			}
+		}
+
+		[Test]
+		public void ClassLibraryNoResources ()
+		{
+			var proj = new MSBuildSdkExtrasProject ();
+			proj.Sources.Remove (proj.Sources.First (s => s.BuildAction == "AndroidResource"));
+			using (var b = CreateDllBuilder (Path.Combine ("temp", TestName))) {
+				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
+				var output = Path.Combine (Root, b.ProjectDirectory, proj.OutputPath, proj.ProjectName + ".dll");
+				FileAssert.Exists (output);
+			}
+		}
+
+		[Test]
+		public void MultiTargeting ()
+		{
+			var proj = new MSBuildSdkExtrasProject ();
+			proj.TargetFrameworks += ";netstandard2.0";
+			proj.Sources.Add (new BuildItem.Source ("MyView") {
+				TextContent = () =>
+@"#if __ANDROID__
+class MyView : Android.Views.View
+{
+	public MyView (Android.Content.Context c) : base (c) { }
+}
+#endif",
+			});
+			using (var b = CreateDllBuilder (Path.Combine ("temp", TestName))) {
+				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
+				var output = Path.Combine (Root, b.ProjectDirectory, proj.OutputPath, proj.ProjectName + ".dll");
+				FileAssert.Exists (output);
```

Comment:
Should we (somehow?) assert that the generated file contains `MyView`, ensuring that `__ANDROID__` is defined as part of the MSBuild.Sdk.Extras build?

Reply:
Under mono, `monodis --typedef ASSEMBLY | grep MyView` would work...

Reply:
I think there is a test in here that uses Mono.Cecil to do something like this, I’ll check.